95.md

```diff
@@ -0,0 +1,91 @@
+NIP-95
+======
+
+File Storage
+------------
+
+`draft` `optional` `author:arthurfranca` `author:Semisol` `author:frbitten` `author:staab`
+
+## Motivation
+
+This NIP defines a way to turn a relay into a storage service.
+The spec doesn't use regular nostr events through websockets for
+storing, requesting nor retrieving data because it would be inneficient.
+
+## Relay Adaptation
+
+Relays wishing to be used as file storage should opt-in by making available
+an https route at `/nip95(/<sha256-file-hash>)` for handling files as described below.
+They may announce support with [NIP-11](11.md) `supported_nips` field.
+Other ways of advertising NIP-95 support may be described later.
+
+## Upload
+
+A file can be uploaded one at a time to `https://relay.domain/nip95` as `multipart/form-data` content type using `POST` method with the file object set to the `file` form data field.
+
+Clients must add a [NIP-98](98.md) `Authorization` header with the encoded `payload` tag set to the base64-encoded 256-bit SHA-256 hash of the file (not the hash of the whole request body).
+If using an html form, use an `Authorization` form data field instead.
+
+Others custom form data fields may be added depending on specific relay support.
+
+The `filename` embedded in the file may not be honored by the relay, which can internally store just the filename as the SHA-256 hash value ignoring extra metadata.
+
+Relays must reject the upload if the `payload` tag value contains an already used SHA-256 hash or
+it isn't the same of the received file.
+
+The upload response is a json as follows:
+
+```js
+{
+  nip95: {
+    // SHA-256 hash. Empty string if unsuccessful
+    x: "719171db19525d9d08dd69cb716a18158a249b7b3b3ec4bbdec5698dca104b7b"
+  },
+  errors: {
+    // Empty array if successful
+    nip95: ['error 1', 'error 2']
+  }
+}
```

Comment:
[bikeshed] I don't think we need `nip95` in the response if it's already in the request path.

Reply:
That's fair